### PR TITLE
fix #18886 crash on ambiguous proc cast

### LINKED ISSUE #18886: Nimc crash on ambiguous proc cast
**Test**
```nim
type Foo = (proc(_: pointer): void)


proc bar(i: ptr[int]) = discard
proc bar(i: ptr[char]) = discard


let fooBar = cast[Foo](bar)
```
**Error**
```nim
Error: internal error: expr(nkClosedSymChoice); unknown node kind
No stack traceback available
```
https://play.nim-lang.org/#ix=3zKF

***
```nim
>nim -v
Nim Compiler Version 1.5.1 [Linux: amd64]
Compiled at 2021-09-19
Copyright (c) 2006-2021 by Andreas Rumpf

git hash: 6cd219c3a38c5f1a0712b0b5d07a3ea4ab02ff74
active boot switches: -d:release
```

### PR DESCRIPTION
**Pull request by bung87**
fix #18886

### CODE REVIEW

File: tests/errmsgs/t18886.nim [hunk 1]
```diff
@@ -0,0 +1,18 @@
+discard """
+  cmd: "nim check --hints:off $file"
+  errormsg: ""
+  nimout: '''
+t18886.nim(18, 24) Error: ambiguous identifier: 'bar' -- use one of the following:
+  t18886.bar: proc (i: ptr int){.noSideEffect, gcsafe, locks: 0.}
+  t18886.bar: proc (i: ptr char){.noSideEffect, gcsafe, locks: 0.}
+'''
```
Comment (Araq) [inline tests/errmsgs/t18886.nim:8]: But you cannot use this syntax instead at all, this error message is highly misleading!

Comment (bung87) [reply tests/errmsgs/t18886.nim:8]: so what's the expected behavior, what I thought is `cast` get ambiguous identifier and the error message point out where they are.

Comment (bung87) [reply tests/errmsgs/t18886.nim:8]: implemente a new feature support choose between two `bar` ?

Comment (Araq) [reply tests/errmsgs/t18886.nim:8]: Make the error message better. Like "ambiguous identifier, you need a helper proc to disambiguate"

**Comment by Araq** [comment]
Awesome, thanks!

**Pull request merged**